lib/browser.ts

```diff
@@ -837,79 +848,85 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
       return 'Protractor.get(' + destination + ') - ' + str;
     };
 
+    let retPromise: wdpromise.Promise<any>;
+    let then = (fn: () => wdpromise.Promise<any>) => {
```

Comment:
Regular promise chain would work, but it's tricky, since the first promise could be one of three different ones, and you want to prematurely exit if `ignoreSynchronization` is on.  In the light of day though, if I move the `ignoreSynchronization` case up, and start the synchronized case with a blank execute, I should be able to make promise chaining work fine.